gateway/gateway.go

```diff
@@ -116,12 +128,18 @@ type Gateway struct {
 	// Defaults to noop.
 	FatalErrorCallback func(err error)
 
+	// OnScalingRequired is the function called, if discord closes with error
+	// code 4011 aka Scaling Required.
+	OnScalingRequired func()
```

Comment:
Is this only called on initial identification or on reconnects as well?

Reply:
This is called whenever Discord closes the WS with 4011, not afterwards. On call, the Gateway will be closed, and the user should signal all other shards to disconnect, and reconnect with more shards.